src/java-opentimelineio/src/main/cpp/io_opentimeline_OTIOFinalizer.cpp

```diff
@@ -0,0 +1,54 @@
+#include <handle.h>
+#include <io_opentimeline_OTIOFinalizer.h>
+#include <utilities.h>
+
+#include <opentimelineio/any.h>
+#include <opentimelineio/version.h>
+
+enum ClassCode {
+    _OpenTimeErrorStatus,
+    _Any,
+    _OTIOErrorStatus,
+};
+
+std::map<std::string, ClassCode> stringToClassCode = {
+        {"io.opentimeline.opentimelineio.Any",         ClassCode::_Any},
+        {"io.opentimeline.opentime.ErrorStatus",       ClassCode::_OpenTimeErrorStatus},
+        {"io.opentimeline.opentimelineio.ErrorStatus", ClassCode::_OTIOErrorStatus},
+};
+
+std::map<ClassCode, std::string> classCodeToString = {
+        {ClassCode::_Any,                 "io.opentimeline.opentimelineio.Any"},
+        {ClassCode::_OpenTimeErrorStatus, "io.opentimeline.opentime.ErrorStatus"},
+        {ClassCode::_OTIOErrorStatus,     "io.opentimeline.opentimelineio.ErrorStatus"},
+};
+
+/*
+ * Class:     io_opentimeline_OTIOFinalizer
+ * Method:    disposeNativeObject
+ * Signature: (JLjava/lang/String;)V
+ */
+JNIEXPORT void JNICALL Java_io_opentimeline_OTIOFinalizer_disposeNativeObject
+        (JNIEnv *env, jobject thisObject, jlong nativeHandle, jstring nativeClassName) {
+    std::string className = env->GetStringUTFChars(nativeClassName, 0);
+    switch (stringToClassCode[className]) {
+        case _Any: {
+            auto obj = reinterpret_cast<OTIO_NS::any *>(nativeHandle);
+            delete obj;
+            break;
+        }
+        case _OpenTimeErrorStatus: {
+            auto obj = reinterpret_cast<opentime::ErrorStatus *>(nativeHandle);
+            delete obj;
+            break;
+        }
+        case _OTIOErrorStatus: {
+            auto obj = reinterpret_cast<OTIO_NS::ErrorStatus *>(nativeHandle);
+            delete obj;
+            break;
+        }
+        default:
+            throwRuntimeException(env, "Could not find class.");
+    }
```

Comment:
It may make sense to have this code null-check `nativeHandle` before casting and doing a `delete` and then assigning `nativeHandle` null after doing the `delete` so the behavior will be idempotent. If you think that the system will be set up in a safe enough way that multiple calls are not going to happen to this, then it's okay to leave it as-is.

Reply:
I thought this too. A null check before deleting makes sense just in case a second call happens somehiet but :

1. I don't think jlong values passed by reference, are they? Making nativeHandle null after deleting, will that affect the handle on the java side?

2. Multiple calls shouldn't happen. Once the delete function is called the phantom reference will be removed from the queue and the java object will be destroyed. But does anyone else see any loophole?